modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js

```diff
@@ -112,30 +194,169 @@ export const Column = ({
 					/>
 				)}
 
+				<ResizableColumn
+					allowNestedFields={allowNestedFields}
+					columnIndex={columnIndex}
+					drag={drag}
+					drop={drop}
+					editable={editable}
+					instanceId={firstField.instanceId}
+					isFieldSelected={isFieldSelected}
+					isFieldSetOrGroup={isFieldSetOrGroup}
+					onResizing={(resizing) => setResizing(resizing)}
+					pageIndex={pageIndex}
+					parentField={parentField}
+					ref={resizeRef}
+					resizeInfoRef={resizeInfoRef}
+					rootParentField={rootParentField}
+					rowIndex={rowIndex}
+					rowRef={rowRef}
+				>
+					{column.fields.map((field, index) =>
+						children({
+							field,
+							index,
+							loc: {
+								columnIndex,
+								pageIndex,
+								rowIndex,
+							},
+						})
+					)}
+				</ResizableColumn>
```

Comment:
@diegonvs there is a coupling here that ends up that it makes no sense to have the `ResizableColumn` separately. Some things I noticed:

- There is a dependency on `drag` and `drop`
- It depends on `DraggableField` which will not help with the resize part

So maybe we are concentrating logic and responsibilities beyond the scope of Resize for this component, we can move a few things to separate these responsibilities and the name of the component to living up to its responsibility will also be easier to maintain.

Maybe just moving `DraggableField` (Read my comment on that component too) here already solves practically the entire liability problem. Consider the following:

```suggestion
				<ResizableColumn
					currentLoc={{columnIndex, pageIndex, rowIndex}}
					disabled={!isFieldSelected || !editable}
					instanceId={firstField.instanceId}
					onResizing={(resizing) => setResizing(resizing)}
					resizeInfoRef={resizeInfoRef}
					rowRef={rowRef}
				>
					<div
						className={classNames('ddm-drag', {
							'py-0': isFieldSetOrGroup
						})}
						ref={(node) => {
							if (allowNestedFields && !rootParentField.ddmStructureId) {
								drag(drop(node));
							}
							else if (!hasFieldSet(parentField)) {
								drag(node);
							}

							resizeRef.current = node;
						}}
					>
						{column.fields.map((field, index) =>
							children({
								field,
								index,
								loc: {
									columnIndex,
									pageIndex,
									rowIndex,
								},
							})
						)}
					</div>
				</ResizableColumn>
```

With that we also removed some dependencies of some properties that `ResizableColumn` had due to the current implementation of this PR. Some things for notes:

- `disabled` is a more friendly API than passing `isFieldSelected` and `editable` to the component which in the end it decides whether to disable the resize handle
- `currentLoc` the final idea is to provide the loc of the current field, so instead of passing each index of the loc as a property to the component, we can only pass the assembled loc since it is the final objective of the component and we managed to decrease the amount of APIs.
- `drag`, `drop`, `allowNestedFields`, `rootParentField`, `parentField`, `isFieldSetOrGroup` and `ref=resizeRef` have been removed since we no longer have the `drag` and `drop` coupling inside the resize.

One thing to keep in mind is, try to create APIs that can help you understand the final purpose of the component, an example of this is passing `disabled` instead offer `isFieldSelected` and `editable`. The big goal is to provide a component that can cover a responsibility well or allow it to be reused.

Reply:
> One thing to keep in mind is, try to create APIs that can help you understand the final purpose of the component, an example of this is passing disabled instead offer isFieldSelected and editable. The big goal is to provide a component that can cover a responsibility well or allow it to be reused.


💖